hoprnet/hoprnet
# Feature/hopr admin v2

This PR supersedes #3647  for better readability of the PR.

Related https://github.com/hoprnet/hoprnet/issues/3383

#### Changelog

- decouple `hopr-admin` and `hoprd`, `hopr-admin` uses API v2 now
- moved commands logic into `hopr-admin` as it is app specific
- introduced URL query parameters for `apiEndpoint` and `apiToken` (using default values, if not provided)
- imporved `Command` abstract class
- converted dApp to `typescript` 
- remove API v1
- removes usage of `--rest*` flags from various scripts
- listen to messages from `stream` and `messages` endpoints
- improve how command usage is asserted
- more consistent logging
- `hoprd --run` now supports less commands, TODO: adapt --run to work with API logic instead of manually adding commands
- cluster script and run:faucet:all adjusted so they work on my machine, but most likely breaks something, please check
- docs: removed mentions of `fund` command

#### todo

- [x] restore `run` flag
- testing
  - [x] implement more tests into `hopr-admin`
  - [x] update E2E tests since API v1 was removed

#### reviewing this PR

This is a massive PR.

- Review non `hopr-admin` code first, `hopr-admin` is not as important.
- Improve fixes in `local-cluster` script and `run:faucet:all`

#### known

- withdraw doesn't work
- sign doesn't work
- once `hopr-admin` loads and connects to the API, it still thinks it's disconnected, it gets fixed on first interaction
- healthcheck still available under `/v1`, using healthcheck flag will enable it

### testing

0. Build: `make deps && make build`
1. Run local cluster: `./scripts/setup-local-cluster.sh -i $(pwd)/scripts/topologies/full_interconnected_cluster.sh`
2. Go to admin of one node, e.g.: `http://localhost:19505/?apiEndpoint=http://localhost:13305&apiToken=^^LOCAL-testing-123^^`
2.1. Verify admin turns green after some time and commands can interact with node via the API
3. Go to status board, e.g.: `https://hopr-status-board.vercel.app/?apiEndpoint=http://localhost:13305&apiToken=^^LOCAL-testing-123^^`
3.1. Verify data from nodes can be displayed, which ensure nothing has been broken due to the upgrade.
